services/web/server/src/simcore_service_webserver/socketio/events.py

```diff
@@ -17,6 +17,23 @@
 
 SOCKET_IO_PROJECT_UPDATED_EVENT: str = "projectStateUpdated"
 SOCKET_IO_NODE_UPDATED_EVENT: str = "nodeUpdated"
+SOCKET_IO_LOG_EVENT: str = "logger"
+
+
+async def send_messages(
+    app: Application, user_id: str, messages: Dict[str, Any]
+) -> None:
+    sio: AsyncServer = get_socket_server(app)
+
+    socket_ids: List[str] = []
+    with managed_resource(user_id, None, app) as rt:
+        socket_ids = await rt.find_socket_ids()
+
+    send_tasks = deque()
+    for sid in socket_ids:
+        for event_name, data in messages.items():
+            send_tasks.append(sio.emit(event_name, json.dumps(data), room=sid))
+    await logged_gather(*send_tasks, reraise=False, log=log, max_concurrency=10)
```

Comment:
THOUGHT: i wonder how the logger plays with async tasks ... i am aware that it is blocking but never had the time to profile it ... but according to https://www.roguelynn.com/words/asyncio-profiling/ it looks like it would be a good idea to switch to some asyncio-friendly logger

Reply:
fully agree